core/commonMain/src/kotlinx/serialization/encoding/Decoding.kt

```diff
@@ -524,10 +524,15 @@ public inline fun <T> Decoder.decodeStructure(
     block: CompositeDecoder.() -> T
 ): T {
     val composite = beginStructure(descriptor)
+    var ex: Throwable? = null
```

Comment:
Wouldn't it be the same just to write

```kotlin
val composite = beginStructure(descriptor)
val result = composite.block()
composite.endStructure()
return result
```

without all try-catch logic?

Reply:
It's going to break for a `return` statement.
And marking it with `crossinline` makes things less pretty